packages/graphql-lowcode/src/generate/roots/roots.ts

```diff
@@ -34,5 +37,43 @@ export function getRoots(types: TypesObject[], rootNames: { type: string, name?:
   const mutationRoot = roots.find(root => root.name === mutation.name)
   const subscriptionRoot = roots.find(root => root.name === subscription.name)
 
-  return[queryRoot, mutationRoot, subscriptionRoot]
+  return [queryRoot, mutationRoot, subscriptionRoot]
+}
+
+export function getQueryNames(introspection: IntrospectionQuery, entityName: string) {
+  const rootNames = getRootNames(introspection)
+  const [queryRoot, mutationRoot, subscriptionRoot] = getRoots(introspection.types, rootNames)
+
+  const listTypeQuery = queryRoot?.fields.find(field => isListType(field)) ?? queryRoot?.fields[0] //TODO zmenit
+  const detailTypeQuery = queryRoot?.fields.find(field => isObjectType(field, entityName) ?? queryRoot?.fields[0])
+
+  //TODO update, insert etc...
+
+  return {
+    getListTypeQueryName: () => listTypeQuery?.name,
+    getDetailTypeQueryName: () => detailTypeQuery?.name
+  }
+}
+
+function isListType(typeField: Field) {
```

Comment:
`for` loop instead of `while` loop for expressing next iteration/increment
```
function isListType(typeField: Field) {
  for (; typeField.type; typeField = typeField.ofType) {
    if (actualType.kind === 'LIST') return true
  }

  return false
```